### ISSUE TITLE
Use SQLite using Room and other Architecture Components

### ISSUE THREAD

**Issue by sriramr98** [avjinder/Minimal-Todo#78]
Using files to save data is not a good way to save notes data. With the new architecture components and Room, we can create better apps with better code quality. I would like to take up this issue and work on this mainly using LiveData, ViewModel and Room

**Comment by avjinder** [comment]
I've already started work on this. I'm doing a complete rewrite in Kotlin, as well as using the arch. components such as Room, LiveData, ViewModel etc.

**Comment by sriramr98** [comment]
It'll also be better if you can make some kind of backend to backup data. I can help you with migrating to room and the backend if needed.

**Comment by avjinder** [comment]
Yeah, that sounds interesting. I'm thinking of backing up to Dropbox, Drive, OneDrive etc. Also Google Tasks integration at some point.
I'd love your help with the Architecture Components, since I'm pretty new at it and my arch. code is a bit rough around the edges.

**Comment by sriramr98** [comment]
Yeah sure. If you can mail me at sriramr083@gmail.com , I have some ideas which can be implemented and provide that as a free service to everyone.
